account_payment_promissory_note/models/account_abstract_payment.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2018 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAbstractPayment(models.AbstractModel):
+    _inherit = 'account.abstract.payment'
+
+    promissory_note = fields.Boolean(
+        string='Promissory Note',
+    )
+    date_due = fields.Date(
+        string='Due Date',
+    )
+
+    @api.onchange('promissory_note')
+    def _onchange_promissory_note(self):
+        if not self.promissory_note:
+            self.date_due = False
+            return self.date_due
+        invoices = False
+        if self._name == 'account.payment':
```

Comment:
This is still incorrect. You must override the method on `account.payment` model and add this logic